_posts/2021-08-20-A query, or There and Back Again.md

```diff
@@ -0,0 +1,67 @@
+
+A **node** is an instance of OpenSearch and a **cluster** is a collection of one or more OpenSearch nodes with the same cluster name. There are several different types of OpenSearch nodes that we can discuss in future blog posts (stay tuned), but at a high-level OpenSearch functionality is built off the distribution of tasks and work among all the nodes in a cluster. OpenSearch allows for indices to be subdivided into multiple **shards** and each shard is part of the OpenSearch index. You can copy those index shards as **replica shards** (or just replicas) which serve as redundant copies of data, increasing both resiliency and capacity for read requests. 
+
+## A Query’s Journey
+
+Now that we have reviewed OpenSearch components, let’s follow a query through OpenSearch. At a very high-level, a OpenSearch query can be broken down into two major phases; the query phase and the fetch phase. 
+
+### Query Phase
+In this phase, the query provided to OpenSearch is broadcasted to a copy of every shard across the entire index. Once received, the query is executed locally. The result is a priority queue of matching, sorted documents for each shard. This priority queue is simply a sorted list of the *top n* matching documents with *top* being determined by relevance and *n* being determined by pagination parameters set by the user (or the default if not set by the user). Relevance in this case is a score of how well each document it matches the query. The individual shards are responsible for the actual matching process as well as the scoring. So for example, if you have a three node cluster and wanted to search for "Hamster", you could write a query like this;
+```
+curl -X GET "localhost:9200/_search?q=Hamster&pretty"
+```
+Once OpenSearch has your query, OpenSearch takes the following steps;
+1. The API or client sends your search query to Node 1. 
+2. Node 1 sends the search request to a primary or replica shard for each shard in an index. 
+3. Each shard executes the search locally and creates a locally sorted queue.
+4. Each shard returns doc ID and sort value of the all the documents in its local queue.
+5. Node 1 as the coordinator node, merges these values into a globally sorted list.
+
+*Figure 1: Query Phase*
+![Figure 1: Query Phase Diagram](https://user-images.githubusercontent.com/9532728/130289482-4ef3379a-1dbf-49b7-af36-ba9339864005.png)
```

Comment:
You can't host your images on github - this will break in production. Put the images in a corresponding subfolder of [assets/media/blog-images](https://github.com/opensearch-project/project-website/tree/main/assets/media/blog-images). You also need to add `{: .img-fluid}` to the end of the image markdown (see other blog posts for example). If you don't the image will bust out of the template:
<img width="1287" alt="Screen Shot 2021-08-23 at 8 50 53 AM" src="https://user-images.githubusercontent.com/1152927/130468580-1121af56-fe1d-4b45-90c2-c5e908cf4462.png">

Reply:
Updated to;

![Figure 1: Query Phase Diagram]({{ site.baseurl }}/assets/media/blog-images/2021-08-23-a-query-or-there-and-back-again/QueryPhase.png){: .img-fluid }

Working to validate on local.